R/mail.R

```diff
@@ -4,238 +4,566 @@
+      } else {
+        if (verbose) {
+          cat("Failed to download:", download_url, "\n")
+          cat("HTTP Status Code:", status_code, "\n")
+        }
+        # Remove failed download file.
+        unlink(file_path)
+        failed_months <- c(failed_months, year_month_str)
+      }
     }
+  }
 
+  ########## Summary of Failed Downloads ##########
+  if (length(failed_months) > 0) {
+    warning("The following months could not be downloaded (no data available or other error):\n", paste(failed_months, collapse = ", "))
   }
 
-  #Close connection to target mbox file
-  close(fileConn)
+  # List the files in the save_folder_path
+  downloaded_files <- list.files(save_folder_path, pattern = "kaiaulu_\\d{6}\\.mbox$", full.names = FALSE)
 
-  #return output location
-  return(output)
-}
+  # Extract the YYYYMM from the file names
+  downloaded_dates <- as.numeric(sub("kaiaulu_(\\d{6})\\.mbox", "\\1", downloaded_files))
 
-#' Compose mod_mbox archives (.mbox) into a single mbox file for use with \code{\link{parse_mbox}}
-#' @param base_url An url pointing to the mod_mbox directory (e.g. "http://mail-archives.apache.org/mod_mbox") without trailing slashes
-#' @param mailing_list Name of the project mailing list (e.g. apr-dev) in the mod_mbox directory
-#' @param from_year First year in the range to be downloaded
-#' @param to_year Last year in the range to be downloaded
-#' @param save_folder_path the full *folder* path where the monthly downloaded mbox will be stored.
-#' @param verbose Prints progress during execution
-#' @return Returns the path of the downloaded mbox file.
-#' @export
-download_mod_mbox_per_month <- function(base_url, mailing_list, from_year, to_year, save_folder_path,verbose=FALSE) {
+  # Find the expected list of YYYYMM between start_year_month and end_year_month
+  start_date <- as.Date(paste0(start_year_month, "01"), format = "%Y%m%d")
+  end_date <- as.Date(paste0(end_year_month, "01"), format = "%Y%m%d")
+  all_dates <- seq(start_date, end_date, by = "month")
+  expected_dates <- as.numeric(format(all_dates, "%Y%m"))
 
+  # Identify missing months
+  missing_months <- setdiff(expected_dates, downloaded_dates)
 
-  #Initialize variables
-  counter <- 0
-  destination <- list()
+  # Determine the earliest and latest dates downloaded
+  if (length(downloaded_dates) > 0) {
+    min_downloaded_date <- min(downloaded_dates)
+    max_downloaded_date <- max(downloaded_dates)
 
-  #Open file handle to output file
-  output <- path.expand(save_folder_path)
+    if (verbose) {
+      cat("\nSummary of Downloads:\n")
+      cat("save_folder_path contains mail from date", min_downloaded_date, "to", max_downloaded_date, "\n")
+    }
+  } else {
+    if (verbose) {
+      cat("No files found in save_folder_path\n")
+    }
+  }
 
-  #Loop through time and compose the mbox file
-  for (year in (from_year:to_year)) {
+  if (length(missing_months) == 0) {
+    if (verbose) {
+      cat("No missing months\n")
```

Comment:
Any reason cat is being used here? I believe Kaiaulu uses print everywhere. Should change for consistency.

Reply:
Replaced with message using example from jira.R